TypeScript/myPermCheck.ts

```diff
@@ -0,0 +1,81 @@
+// @ts-nocheck
+/**
+ * This is perm check, it allows users to parse the permission you want and let the plugin do the rest. (check bot or user for that perm).
+ *
+ * @author @Benzo-Fury [<@762918086349029386>]
+ * @author @needhamgary [<@342314924804014081>]
+ * @version 1.1.0
+ * @example
+ * ```ts
+ * import { requirePermission } from "../plugins/myPermCheck";
+ * import { commandModule } from "@sern/handler";
+ * export default commandModule({
+ *  plugins: [ requirePermission('target', 'permission', 'No permission response') ],
+ *  execute: (ctx) => {
+ * 		//your code here
+ *  }
+ * })
+ * ```
+ */
+
+import {
+  type GuildMember,
+  PermissionResolvable,
+  GuildMemberResolvable,
+} from "discord.js";
+import { CommandType, EventPlugin, PluginType } from "@sern/handler";
+export function requirePermission(
+  target: GuildMemberResolvable,
+  perm: PermissionResolvable,
+  response?: string
+): EventPlugin<
```

Comment:
view my comment, but basically you can just do `): EventPlugin` and it'll support all command types

Reply:
Oh I didn't realize that. I'll do that.

Reply:
When I set it to that, it throws errors within the command saying it's not assignable to Modals, so I'll change it back unless you find a better way and I'll just put Text in too.